src/runtime/security/negotiation_service.cpp

```diff
@@ -44,10 +48,42 @@ void negotiation_service::on_negotiation_request(negotiation_rpc rpc)
         return;
     }
 
-    server_negotiation *srv_negotiation =
-        static_cast<server_negotiation *>(rpc.dsn_request()->io_session->get_negotiation());
+    server_negotiation *srv_negotiation = nullptr;
+    {
+        utils::auto_lock<utils::ex_lock_nr> l(_lock);
+        srv_negotiation =
```

Comment:
Is it safe to add an `assert` to check the session is not exist in the map?

Reply:
Yes, it's safe, we can add a `assert` here. Because only a `rpc_session` who have a corresponding `negotiation` can send negotiation message